### PR TITLE
Add hex and octel support to Hy integers

### PR DESCRIPTION
**Pull request by paultag**
This allows us to parse things like 0xDEADBEEF or 0o080.

Filed as issue #937

### CODE REVIEW

**Comment by paultag** [comment]
Easy peasy changeset. Who's looking for easy karma, @hylang/core ? :)

:confetti_ball:

**Comment by theanalyst** [comment]
LGTM

**Comment by algernon** [comment]
Looks good to me too.

**Comment by farhaven** [comment]
Looks good. Do we want additional documentation for this or do we assume people familiar with Python know how to specify these integers?

File: tests/native_tests/language.hy [hunk 1]
```diff
@@ -12,6 +12,16 @@
   (assert (isinstance sys.argv list)))
 
 
+(defn test-hex []
+  "NATIVE: test hex"
+  (assert (= 0x80 128)))
+
+
+(defn test-octel []
```
Comment (zackmdavis) [inline tests/native_tests/language.hy]: (The standard spelling [is](https://en.wikipedia.org/wiki/Octal) *octal*, with an *a*.)

**Comment by paultag** [comment]
'octel' was spelled wrong. It's 'octal', Way to go 3 AM me. I also added `0b` as a leader for binary. Thanks @olasd

**Comment by paultag** [comment]
@farhaven Hurm, good question. Thoughts?

File: tests/native_tests/language.hy (hunk 1 above)
Comment (paultag) [inline tests/native_tests/language.hy]: @zackmdavis thanks! Yeah, only just noticed this. Fixed, thank you!

**Comment by paultag** [comment]
My guess is folks who need to write hex would know to write `0x`. `0o` is a weird one to be sure, but my guess is folks would go for how Python does it first in Hy. Not opposed to documenting it, but I'm on the 'meh' side of the fence. Other folks?

**Comment by kirbyfan64** [comment]
Doesn't Python 2 still use the other octals though: http://stackoverflow.com/questions/11620151/what-do-numbers-starting-with-0-mean-in-python.

**Comment by paultag** [comment]
Yes, but it understands this notation:

```
[paultag@cassiel:~][⌚ 11:33 AM] ♥  python2
Python 2.7.10 (default, Jul  1 2015, 10:54:53)
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> int("0o13424", base=8)
5908
>>>
```

So this will work fine :)

On Tue, Sep 15, 2015 at 11:27 AM, Ryan Gonzalez <notifications@github.com>
wrote:

> Doesn't Python 2 still use the other octals though:
> http://stackoverflow.com/questions/11620151/what-do-numbers-starting-with-0-mean-in-python
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hylang/hy/pull/938#issuecomment-140429975>.
>



-- 
:wq

**Comment by kirbyfan64** [comment]
@paultag I was using that as an example to why it should be documented. :)

**Comment by paultag** [comment]
Ah! A good point!

**Comment by paultag** [comment]
Quick note added to docs. Nothing fancy.

**Comment by farhaven** [comment]
With the added docs, I'd say LGTM. Who wants to do the honors of clicking the "merge" button?

**Comment by paultag** [comment]
@hylang/core - got two acks who's doin' it? :)

**Comment by berkerpeksag** [comment]
I'm the chosen one :)

**Comment by paultag** [comment]
Thank you! :D

On Wed, Sep 16, 2015 at 1:56 PM, Berker Peksag <notifications@github.com>
wrote:

> I'm the chosen one :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hylang/hy/pull/938#issuecomment-140821486>.
>



-- 
:wq

**Pull request merged**